src/classes/Entity/types.ts

```diff
@@ -345,10 +345,15 @@ export type PutItem<MethodItemOverlay extends Overlay,
   EntityItemOverlay,
   A.Compute<CompositePrimaryKey &
     O.Pick<Item, Attributes['always']['input'] | Attributes['required']['input']> &
-    O.Partial<O.Pick<Item,
-      | Attributes['always']['default']
-      | Attributes['required']['default']
-      | Attributes['optional']>>>
+    O.Partial<{
+      [inputAttr in Attributes['always']['input'] | Attributes['required']['input']]:
+      | Item[A.Cast<inputAttr, keyof Item>]
+    } &
+    {
+      [attr in Attributes['optional']]?:
```

Comment:
You're already on a partial type, why still use `[attr in Attributes['optional']]?`?

Reply:
Hope it's clearer now @fredericbarthelet  @ThomasAribart. Also added tests to cover allowing null types for optional attributes ! :)